realm/realm-library/src/main/java/io/realm/RealmSchema.java

```diff
@@ -221,20 +221,38 @@ Table getTable(String className) {
     Table getTable(Class<? extends RealmModel> clazz) {
         Table table = classToTable.get(clazz);
         if (table == null) {
-            clazz = Util.getOriginalModelClass(clazz);
-            table = transaction.getTable(realm.configuration.getSchemaMediator().getTableName(clazz));
-            classToTable.put(clazz, table);
+            Class<? extends RealmModel> originalClass = Util.getOriginalModelClass(clazz);
+            if (originalClass != clazz) {
+                table = classToTable.get(originalClass);
+            }
+            if (table == null) {
+                table = transaction.getTable(realm.configuration.getSchemaMediator().getTableName(originalClass));
+                classToTable.put(originalClass, table);
+            }
+            if (originalClass != clazz) {
```

Comment:
Since `originalClass != clazz` is used twice i would abstract it into `boolean isProxyClass`. I would also make the code more readable.